src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -223,10 +258,69 @@ public Command followTrajectory(String name, PathPlannerTrajectory trajectory){
         ).beforeStarting(() -> trajectoryLog.append("Following trajectory " + name)
         ).andThen(() -> trajectoryLog.append("Trajectory " + name +  " Ended"));
     }
-    
+
+    public Command alignToTarget(NodePosition nodePosition){
+        Integer id = Photonvision.getAprilTagID();
+        Translation2d offset = new Translation2d(0,0); // TODO: calculate offsets
+
+        if(id != -1 && Robot.isRed ? id <= 4 : id > 4){
+            //Get pose of node to be targeted
+            Pose2d desiredPose = new Pose2d(
+                nodePosition.getTranslation(id).plus(offset),
+                Robot.isRed ? Rotation2d.fromDegrees(0) : Rotation2d.fromDegrees(180)
+            );
+            //Generate trajectory to desired pose
+            Pose2d currentPose = getPose();
+            return followTrajectory(
+                "Align",
+                AutonSubsytem.generateTrajectory(
+                    AutonSubsytem.constructPoint(
+                        currentPose, 
+                        Units.radiansToDegrees(
```

Comment:
Shouldn't the starting point of the trajectory have the actual heading of the currentPose?

Reply:
Poses do not contain a heading. I think you're confusing holonomic rotation and heading, rotation is what direction the robot is physically pointed in (contained in current pose), heading is only used to alter the behavior of pathplanner trajectories